c7n/resources/asg.py

```diff
@@ -574,9 +574,10 @@ def process(self, asgs, event=None):
 
     def get_resource_date(self, asg):
         cfg = self.launch_info.get(asg)
-        ami = self.images.get(cfg.get('ImageId'), {})
-        return parse(ami.get(
-            self.date_attribute, "2000-01-01T01:01:01.000Z"))
+        if cfg:
```

Comment:
Returning `None` if cfg is not found for safety reasons. Let me know if you think otherwise @kapilt . Thanks!!

Reply:
ideally we return the parsed default date across scenarios on fallback.. if launch info is a dictionary then cfg = self.launch_info.get(asg, {}) suffices and can remove this conditional

Reply:
self.launch_info is not a dict here. Its of type `<class 'c7n.resources.asg.LaunchInfo'>`. Doing cfg = self.launch_info.get(asg, {}) give a TypeError: get() takes 2 positional arguments but 3 were given.

If we want cfg = {} on fallback, I can do:
```
if cfg is None:
   cfg = {}
```
If this makes sense to you! Thanks

Reply:
the reason for the suggestion re doing a default dict value for config, is that this is otherwise returning a new value None from this method, where as previously it took care to always return a default date.  sounds good re conditional

Reply:
yep, makes complete sense @kapilt . I have addressed it. Thanks much!!